packages/core/src/api/apis/default/ErrorApiForwarder.ts

```diff
@@ -0,0 +1,26 @@
+import { ErrorApi, ErrorContext } from '../../../';
+import { Color } from '@material-ui/lab';
+
+type SubscriberFunc = (error: Error) => void;
+type Unsubscribe = () => void;
+
+export class ErrorApiForwarder implements ErrorApi {
```

Comment:
I think it would be cleaner to invert the dependencies and make this depend on the alert API instead. Calling it something like `AlertingErrorApi`

Reply:
hmm but when you post an error you would want to trigger an alert not the other way around. the alert display needs its api implementation to be the alerting api and by default it subscribes to the error api.

Reply:
I'm thinking this implementation of the error api would be something like

```tsx
post(error: Error, context?: ErrorContext) {
  if (!context?.hidden) {
    this.alertApi.post({
      body: String(error),
      severity: 'error',
    });
  }
}
```

Reply:
oh i see, i felt that the error api shouldn't really know anything about alerting. this way the alerting api can just subscribe to the various apis that it wants to alert on. seemed a clearer separation of concerns. but if you feel strongly about it i can change

Reply:
This isn't the error api though, it's an implementation of it, so we can really do whatever we want here as long as we adhere to the ErrorApi type. It's up to the app/test wrappers to pick and choose the implementation. The best example of that in use I think is our internal user api, where you have a couple of different implementations depending on if we're in prod/override/tests.

Reply:
sure, but i think its really the separation of concerns that i care about. the alerting implementation should care about whats being alerted on. the error api has nothing to do with alerting

Reply:
if we wanted to rip out alerting for example, we'd have to update all the other apis that are calling it vs just removing its implementation

Reply:
@Rugvip ok well in any case it's up to you so just let me know how you'd like to proceed

Reply:
@wesleyyee since you'd have to add the subscribe function to the ErrorApi to be able to subscribe from the AlertApi implementation, I'd rather go with the ErrorApi calling the AlertApi directly.

Regarding the design my view is that the AlertApi impl shouldn't care about who wants to show alerts, it just provides a way to show them. While this ErrorApi impl chooses to handle errors by showing them as alerts.

Reply:
done